DittoHeartbeat/src/main/java/live.ditto.dittoheartbeat/Heartbeat.kt

```diff
@@ -0,0 +1,129 @@
+package live.ditto.dittoheartbeat
+
+import live.ditto.Ditto
+import android.os.Handler
+import android.os.Looper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.launch
+import live.ditto.DittoConnectionType
+import live.ditto.DittoPeer
+import java.math.BigInteger
+import java.security.MessageDigest
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+
+data class HeartbeatConfig(
+    val id: Map<String, String>,
+    val interval: Long,
+    val collectionName: String,
+)
+
+data class HeartbeatInfo(
+    val id: Map<String, String>,
+    val lastUpdated: String,
+    val presence: Presence?,
+)
+
+data class Presence(
+    val remotePeersCount: Int,
+    val peers: List<DittoPeer>,
+)
+
+var presence: Presence? = null
+
+fun startHeartbeat(ditto: Ditto, config: HeartbeatConfig): Flow<HeartbeatInfo> = flow {
+    while (true) {
+        delay(config.interval)
+        val timestamp = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.getDefault()).format(Date())
+        val info = HeartbeatInfo(
+            id = createCompositeId(config.id, ditto),
+            lastUpdated = timestamp,
+            presence = observePeers(ditto),
+        )
+        addToCollection(info, config, ditto)
+        emit(info)
+    }
+}
+
+fun createCompositeId(configId: Map<String, String>, ditto: Ditto): Map<String, String> {
+    val compositeId: MutableMap<String, String> = configId.toMutableMap()
+    val presenceGraph = ditto.presence.graph
+    compositeId["dittoPeerKey"] = byteArrayToHash(presenceGraph.localPeer.peerKey)
```

Comment:
Assuming this is the same format, should we add the `pk` prefix here to align these to the keys used in Small Peer Info and elsewhere?

Reply:
I had it as `dittoPeerKey` since this will be added to their composite Id automatically. I wanted to give visibility that it was a ditto variable. Didn't know if users would be confused if it was just `pk` and why it was automatically being added to their composite Id.

Reply:
Yeah let's try to align this with the `pk` prefix